matrix_client/client.py

```diff
@@ -112,6 +112,23 @@ def set_sync_token(self, token):
     def set_user_id(self, user_id):
         self.user_id = user_id
 
+    def register_as_guest(self):
+        """ Register a guest account on this HS.
+        Note: HS must have guest registration enabled.
+        Returns:
+            str: Access Token
+        Raises:
+            MatrixRequestError
+        """
+        response = self.api.register(query_params={'kind': 'guest'})
+        self.user_id = response["user_id"]
+        self.token = response["access_token"]
+        self.hs = response["home_server"]
+        self.api.token = self.token
+        self.sync_filter = '{ "room": { "timeline" : { "limit" : 20 } } }'
```

Comment:
This was just following what was already being done in the code-base (see `register_with_password`), I don't mind changing it though.